### PR TITLE
Coredns update to v1.8.0

### PR DESCRIPTION
**Pull request by ericgraf**
Upversion to upstream coredns version 1.8.1

Release notes: 
- [https://coredns.io/2020/10/22/coredns-1.8.0-release/](https://coredns.io/2020/10/22/coredns-1.8.0-release/)
- [https://coredns.io/2021/01/20/coredns-1.8.1-release/](https://coredns.io/2021/01/20/coredns-1.8.1-release/)

https://github.com/giantswarm/coredns-app/pull/69 and https://github.com/giantswarm/coredns-app/pull/70 needs to be merged and coredns-app needs to be versioned for each one before this change can be merged

### CODE REVIEW

**Comment by stone-z** [comment]
We should probably link to the 1.8.0 release notes here as well since breaking changes are included there
https://coredns.io/2020/10/22/coredns-1.8.0-release/

**Comment by stone-z** [comment]
Reverting to coredns 1.8.0 because upstream hasn't published a tagged 1.8.1 image yet https://github.com/coredns/coredns/issues/4418

**Comment by stone-z** [comment]
Tested on `ginger` and `ghost` -- workloads clusters pass CNCF conformance with k8s 1.18 and `coredns` 1.8.0

**Comment by stone-z** [comment]
@giantswarm/team-ludacris PTAL

File: helm/coredns-app/Chart.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 apiVersion: v1
-appVersion: 1.7.1
+appVersion: 1.8.1
```
Comment (MarcelMue) [inline helm/coredns-app/Chart.yaml:2]: Wrong version or am I missing some essential magic?

Comment (stone-z) [reply helm/coredns-app/Chart.yaml:2]: Yeah that's the wrong version. Update inbound

**Review by MarcelMue — APPROVED** [review]
LGTM

**Pull request merged**